web-services-app/src/test/java/edu/unc/lib/boxc/web/services/rest/IiifV3ManifestControllerTest.java

```diff
@@ -141,4 +144,72 @@ public void testGetCanvas() throws Exception {
+                .andExpect(status().isOk())
+                .andReturn();
+
+        Map<String, Object> respMap = MvcTestHelpers.getMapFromResponse(result);
+        assertEquals("Manifest", respMap.get("type"));
+        assertEquals("http://example.com/iiif/v3/f277bb38-272c-471c-a28a-9887a1328a1f/manifest", respMap.get("id"));
+        assertEquals("Test Work", ((List) ((Map) respMap.get("label")).get("none")).get(0));
+        var metadata = (List) respMap.get("metadata");
+        assertFalse(metadata.isEmpty());
+    }
+
+    @Test
+    public void testGetCanvasWithVideoFile() throws Exception {
+        var fileObj = new ContentObjectSolrRecord();
+        fileObj.setId(OBJECT_ID);
+        fileObj.setResourceType(ResourceType.File.name());
+        fileObj.setTitle("File Object");
+        var originalDs = new DatastreamImpl("original_file|video/mp4|video.mp4|mp4|0|||240x750x500");
+        fileObj.setDatastream(List.of(originalDs.toString()));
+        when(accessCopiesService.listViewableFiles(eq(OBJECT_PID), any())).thenReturn(List.of(fileObj));
+
+        var result = mockMvc.perform(get("/iiif/v3/" + OBJECT_ID + "/canvas")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        var respJson = MvcTestHelpers.getResponseAsJson(result);
+        var body = respJson.get("items").get(0).get("items").get(0).get("body");
+        assertEquals(SERVICES_BASE + "file/" + OBJECT_ID, body.get("id").textValue());
+        assertEquals("video/mp4", body.get("format").textValue());
+        assertEquals("Video", body.get("type").textValue());
+        assertEquals(750, body.get(WIDTH).intValue());
+        assertEquals(240, body.get(HEIGHT).intValue());
+        assertEquals(500, body.get(DURATION).intValue());
+    }
+
+    @Test
+    public void testGetCanvasWithAudioFile() throws Exception {
+        var fileObj = new ContentObjectSolrRecord();
+        fileObj.setId(OBJECT_ID);
+        fileObj.setResourceType(ResourceType.File.name());
+        fileObj.setTitle("File Object");
+        var originalDs = new DatastreamImpl("original_file|audio/mp4|sound.mp3|mp3|0|||xx500");
+        fileObj.setDatastream(List.of(originalDs.toString()));
+        when(accessCopiesService.listViewableFiles(eq(OBJECT_PID), any())).thenReturn(List.of(fileObj));
+
+        var result = mockMvc.perform(get("/iiif/v3/" + OBJECT_ID + "/canvas")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        var respJson = MvcTestHelpers.getResponseAsJson(result);
+        var body = respJson.get("items").get(0).get("items").get(0).get("body");
+        assertEquals(SERVICES_BASE + "file/" + OBJECT_ID, body.get("id").textValue());
+        assertEquals("Sound", body.get("type").textValue());
+        assertEquals(500, body.get(DURATION).intValue());
+    }
```

Comment:
Based on the other comments, we'll probably want to add test cases for records without extent, and -1 duration.